### ISSUE TITLE
Q: Is there a way to apply this to all public and only public methods of type?

### ISSUE THREAD

**Issue by epitka** [vescon/MethodBoundaryAspect.Fody#86]
I want to do some post processing on every public method invocation of an aggregate. Is there a way to filter application of this?  If I apply to a class level it gets invoked for all methods, even on properties.

**Comment by Ralf1108** [comment]
Currently there is no possibility to filter for specific methods. This has to be implemented first.

Filters could then be:

Methodname via RegEx
By accessibility modifier
By type (method or property)


Another possibility would be to use the approach from PostSharp and CompileTimeValidate method. But this requires executing code from the aspect during weaving.

**Comment by psaxton** [comment]
@epitka (Acknowledging this old issue may no longer be of interest to you.) This may have been solved in v2.0.139 using the MulticastAttributes overload of the attribute. See [src/MethodBoundaryAspect.Fody.UnitTests.TestAssembly.NetFramework/PublicMethodAttributeAspectMethods.cs](https://github.com/vescon/MethodBoundaryAspect.Fody/blob/master/src/MethodBoundaryAspect.Fody.UnitTests.TestAssembly.NetFramework/PublicMethodAttributeAspectMethods.cs) for an example of how this works.

Acknowledging this old issue may no longer be of interest to you, if this is not what you're trying to accomplish, could you please provide a description of what you're wanting to do differently?

**Comment by epitka** [comment]
Yes,  I would like to exclude any properties methods. I wand to add some out-of-box tracing, and I only want user defined methods to be included in trace.

**Comment by JinsPeter** [comment]
@epitka you can just use MulticastAttributes.Public in the Attribute parameters list..
If this solves your requirement, plz close this issue.
